Runigram.java

```diff
@@ -121,21 +225,49 @@ public static Color blend(Color c1, Color c2, double alpha) {
 	 * and (1 - alpha) part the second image.
 	 * The two images must have the same dimensions.
 	 */
-	public static Color[][] blend(Color[][] image1, Color[][] image2, double alpha) {
-		//// Replace the following statement with your code
-		return null;
+	public static Color[][] blend(Color[][] image1, Color[][] image2, double alpha)
+	{
+		int numRows = image1.length;
+		int numCols = image1[0].length;
+		Color[][] blendedImage = new Color[numRows][numCols];
+
+		for (int row = 0; row < numRows; row++)
+		{
+			for (int col = 0; col < numCols; col++)
+			{
+				blendedImage[row][col] = blend(image1[row][col], image2[row][col], alpha);
+			}
+		}
+
+		return blendedImage;
 	}
 
+
 	/**
 	 * Morphs the source image into the target image, gradually, in n steps.
 	 * Animates the morphing process by displaying the morphed image in each step.
 	 * Before starting the process, scales the target image to the dimensions
 	 * of the source image.
 	 */
-	public static void morph(Color[][] source, Color[][] target, int n) {
-		//// Replace this comment with your code
+	public static void morph(Color[][] source, Color[][] target, int n)
+	{
+		int sourceHeight = source.length;
+		int sourceWidth = source[0].length;
+		double alpha = 0;
+
+		Color[][] morphedDimension = scaled(target, sourceWidth, sourceHeight);
+		Color[][] mixed;
+
+		for (int i = 0; i <= n; i++)
+		{
+			alpha = (n - i) / n;
+			mixed = blend(source, morphedDimension, alpha);
+			display(mixed);
+			StdDraw.pause(500);
+
+		}

```

Comment:
Declare variables (alpha) near it's declaration. Read Java Style Guidelines in the moodle.
alpha = (n - i) / n is incorrect. (n-i)/n will allways return 0 because n and i are integers (not float)
4/5 = 0
4/5.0 = 0.8
So the solution can be (n-i) / (double) n